lib/sqlalchemy/dialects/mysql/mariadb.py

```diff
@@ -5,15 +5,24 @@
 # This module is part of SQLAlchemy and is released under
 # the MIT License: https://www.opensource.org/licenses/mit-license.php
 # mypy: ignore-errors
-from .base import MariaDBIdentifierPreparer
+from .base import MariaDBIdentifierPreparer, MySQLTypeCompiler
 from .base import MySQLDialect
 
 
+class MariaDBTypeCompiler(MySQLTypeCompiler):
+    def visit_uuid(self, type_, **kw):
+        if self.dialect.server_version_info >= (10, 7) and type_.native_uuid:
```

Comment:
I would reverse the conditional; default to visit_UUID assuming more modern versions of mariadb, and then if server_version_info < (10, 7) do visit_uuid().   server_version_info can also be None if the dialect was not connected to a database.

Reply:
Not sure if I understand this correctly.
I believe  
def visit_uuid is for Uuid
and
def visit_UUID (from MySQLTypeCompiler) is for UUID
Is it correct? or should I somehow manage relation from visit_uuid (visit_UUID) to data types Uuid (UUID)  or the priority of  visit_uuid and calls?

Reply:
`
    def visit_uuid(self, type_, **kw):
        result = self.visit_UUID(type_, **kw)
        if self.dialect.server_version_info < (10, 7) or not type_.native_uuid:
            result = self.visit_uuid(type_, **kw)
        return result
`
please check if I understand your point correctly